### PR TITLE
Add more friendly error message for policy object

### PR DESCRIPTION
**Pull request by lunixoid**
In our projects we use minio as the storage sever. Each bucket equals - git project. After building project CI uploads artefact into minio and sets `mc policy download` for subdirectory. Name of subdirectory in bucket depends on name of current brunch (develop, master).
So after some time of working our bucket policy file reach limit 20K and send us error:
`Your proposed upload exceeds the maximum allowed object size.`

We spent a lot of time to understand and resolve this issue, so I suggest to extend error message for this case.

### CODE REVIEW

**Comment by minio-ops** [comment]
```
Error running mint automation
+ mkdir -p 7412-6799ae4/gopath/src/github.com/minio
+ git -C 7412-6799ae4/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 7412-6799ae4/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 7412-6799ae4/gopath/src/github.com/minio/minio fetch --quiet minio pull/7412/head:pr7412
+ git -C 7412-6799ae4/gopath/src/github.com/minio/minio checkout --quiet pr7412
+ GOPATH=/mnt/nvme/mint-auto/7412-6799ae4/gopath
+ make -C 7412-6799ae4/gopath/src/github.com/minio/minio --quiet
# github.com/minio/minio/cmd
cmd/api-errors.go:405:75: syntax error: unexpected newline, expecting comma or }
make: *** [build] Error 2

```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#5ad67fa1-398a-42e3-9add-6c19d73db59e) |
| mint-dist-xl.sh | :x: [more...](#68fb3fb6-e76d-4411-bc7f-cd17c0652505) |
<a name="5ad67fa1-398a-42e3-9add-6c19d73db59e"></a>
### 7412-9c416c3/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:31272
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp b7ddb6bf6b76:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-l9i3ck5a1o9cpj",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 391,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.52:31272/aws-sdk-go-test-l9i3ck5a1o9cpj: dial tcp 72.28.97.52:31272: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="68fb3fb6-e76d-4411-bc7f-cd17c0652505"></a>
### 7412-9c416c3/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:31265
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 4b554f974773:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 50 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 32 seconds
(8/14) Running minio-dotnet tests ... done in 33 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 12 seconds
(10/14) Running minio-java tests ... done in 2 minutes and 1 seconds
(11/14) Running minio-js tests ... done in 38 seconds
(12/14) Running minio-py tests ... FAILED in 3 minutes and 8 seconds
{
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1220, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1885, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1228, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "message": "AccessDenied: message: Access Denied",
  "duration": 112,
  "args": {
    "bucket_name": "minio-py-test-db7cac5b-bcb4-45e4-8650-5742b7806add",
    "object_name": "01f29096-e1ce-4ec4-9338-5b3752c8d291"
  },
  "status": "FAIL"
}

Executed 11 out of 14 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#0857e21a-bfe0-46e4-9803-0a002b5d8852) |
| mint-xl.sh | :x: [more...](#3eaa72de-e7f6-4ae9-b750-42d6d78d1280) |
<a name="0857e21a-bfe0-46e4-9803-0a002b5d8852"></a>
### 7412-820626e/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:31720
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 33eb475b1f0d:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-o3y1onx1d11d4u",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 352,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.61:31720/aws-sdk-go-test-o3y1onx1d11d4u: dial tcp 72.28.97.61:31720: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="3eaa72de-e7f6-4ae9-b750-42d6d78d1280"></a>
### 7412-820626e/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:30701
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1b5438768741:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 0 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 48 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 18 seconds
(8/14) Running minio-dotnet tests ... done in 26 seconds
(9/14) Running minio-go tests ... done in 33 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 44 seconds
(11/14) Running minio-js tests ... done in 36 seconds
(12/14) Running minio-py tests ... FAILED in 3 minutes and 54 seconds
{
  "message": "AccessDenied: message: Access Denied",
  "status": "FAIL",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "args": {
    "bucket_name": "minio-py-test-9ab88f7e-dd73-4262-98af-09eaa91afb26",
    "object_name": "b11083e5-3951-45d2-a321-93a87c3bdd2f"
  },
  "duration": 102,
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1220, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1885, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1228, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "name": "minio-py:test_presigned_get_object_expiry_5sec"
}

Executed 11 out of 14 tests successfully.
```

**Comment by poornas** [comment]
@lunixoid, trying to understand why you need to give download access on each individual subdirectory prefix when you might want to just give download access on bucket/* which would in essence accomplish the same without policy bloat.

File: cmd/api-errors.go [hunk 1]
```diff
@@ -399,6 +400,11 @@ var errorCodes = errorCodeMap{
 		Description:    "Your proposed upload exceeds the maximum allowed object size.",
 		HTTPStatusCode: http.StatusBadRequest,
 	},
+	ErrPolicyTooLarge: {
+		Code:           "PolicyTooLare",
```
Comment (harshavardhana) [inline cmd/api-errors.go]: Typo needs to be fixed

File: cmd/api-errors.go [hunk 2]
```diff
@@ -399,6 +400,11 @@ var errorCodes = errorCodeMap{
...
+		Description:    "Policy object exceeds the maximum allowed object size.",
```
Comment (harshavardhana) [inline cmd/api-errors.go]: We should say policy size not object size..

File: cmd/typed-errors.go [hunk 3]
```diff
@@ -38,6 +38,9 @@ var errSizeUnspecified = errors.New("Data size is unspecified")
 // When upload object size is greater than 5G in a single PUT/POST operation.
 var errDataTooLarge = errors.New("Object size larger than allowed limit")
 
+// When policy object size is greater than 20K
+var errPolicyTooLarge = errors.New("Policy object size larger than allowed limit")
```
Comment (harshavardhana) [inline cmd/typed-errors.go]: policy object size is not correct, it should be just policy document size

File: cmd/api-errors.go [hunk 4]
```diff
@@ -1500,6 +1506,8 @@ func toAPIErrorCode(ctx context.Context, err error) (apiErr APIErrorCode) {
 		apiErr = ErrInvalidRange
 	case errDataTooLarge:
 		apiErr = ErrEntityTooLarge
+	case errPolicyTooLarge:
```
Comment (harshavardhana) [inline cmd/api-errors.go]: You don't need this change in this PR since this error is not used anywhere..

File: cmd/object-api-errors.go [hunk 5]
```diff
@@ -76,7 +76,7 @@ func toObjectErr(err error, params ...string) error {
 				Object: params[1],
 			}
 		}
-	case errDataTooLarge:
+	case errDataTooLarge, errPolicyTooLarge:
```
Comment (harshavardhana) [inline cmd/object-api-errors.go]: You don't need this change in this PR since this error is not used anywhere..

File: cmd/typed-errors.go (hunk 3 above)
Comment (harshavardhana) [reply cmd/typed-errors.go]: You don't need this change in this PR since this error is not used anywhere....

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Thanks for the PR, some review comments..

File: cmd/api-errors.go (hunk 1 above)
Comment (lunixoid) [reply cmd/api-errors.go]: Typo fixed.

File: cmd/api-errors.go (hunk 2 above)
Comment (lunixoid) [reply cmd/api-errors.go]: Changed into `document size`

File: cmd/typed-errors.go (hunk 3 above)
Comment (lunixoid) [reply cmd/typed-errors.go]: Removed

File: cmd/api-errors.go (hunk 4 above)
Comment (lunixoid) [reply cmd/api-errors.go]: Agree, removed.

File: cmd/object-api-errors.go (hunk 5 above)
Comment (lunixoid) [reply cmd/object-api-errors.go]: Removed

**Comment by lunixoid** [comment]
> @lunixoid, trying to understand why you need to give download access on each individual subdirectory prefix when you might want to just give download access on bucket/* which would in essence accomplish the same without policy bloat.

It was my mistake, but in any case current error message don't help to understand the problem.
I did it because `mc` prints links to the each file in bucket only if I set policy download for each subdirectory. I use those links for auto downloading files while CI stages.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#a5a8721c-dff1-4fa2-85ca-67207614f885) |
| mint-dist-xl.sh | :x: [more...](#56260725-4f6f-4f41-a293-d0e7513ee82d) |
<a name="a5a8721c-dff1-4fa2-85ca-67207614f885"></a>
### 7412-c7acc2f/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:31540
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a4b7c6ce5d3f:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-1z5ecjsc6sulfy",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 374,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.59:31540/aws-sdk-go-test-1z5ecjsc6sulfy: dial tcp 72.28.97.59:31540: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="56260725-4f6f-4f41-a293-d0e7513ee82d"></a>
### 7412-c7acc2f/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:30272
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 40b833bad76d:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-d1rnappxav9h1j",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 330,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.56:30272/aws-sdk-go-test-d1rnappxav9h1j: dial tcp 72.28.97.56:30272: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#c0e093d3-99a6-4e7a-9056-667c0736d6bb) |
| mint-large-bucket.sh | :x: [more...](#20dd593d-0873-45e4-a3e1-62b95e80da20) |
| mint-compression-dist-xl.sh | :x: [more...](#1c0a212d-8cff-42e1-a9dc-550977893622) |
| mint-compression-fs.sh | :x: [more...](#4a03f820-0de2-4e1c-b4d8-a3a78001956d) |
| mint-fs.sh | :x: [more...](#1da10220-53cf-4c75-9fe1-564915b8b177) |
| mint-dist-xl.sh | :x: [more...](#af1283be-388a-49c8-9d11-6e94d97a5dcf) |
<a name="1c0a212d-8cff-42e1-a9dc-550977893622"></a>
### 7412-6799b54/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:31093
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp cb41f496dcaa:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-qutpjfiusdk9zq",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 2064,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:31093/aws-sdk-go-test-qutpjfiusdk9zq: dial tcp 72.28.97.57:31093: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="c0e093d3-99a6-4e7a-9056-667c0736d6bb"></a>
### 7412-6799b54/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:30305
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7a880660d752:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-whdpm3nch6u6mj",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 3049,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:30305/aws-sdk-go-test-whdpm3nch6u6mj: dial tcp 72.28.97.57:30305: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="1da10220-53cf-4c75-9fe1-564915b8b177"></a>
### 7412-6799b54/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:32016
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 96a75c2f9849:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-x6sq2oh1jcukdf",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 2641,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:32016/aws-sdk-go-test-x6sq2oh1jcukdf: dial tcp 72.28.97.57:32016: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="20dd593d-0873-45e4-a3e1-62b95e80da20"></a>
### 7412-6799b54/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:31039
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7dd984d725fb:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-11zbcl2dprlcea",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 341,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.51:31039/aws-sdk-go-test-11zbcl2dprlcea: dial tcp 72.28.97.51:31039: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="af1283be-388a-49c8-9d11-6e94d97a5dcf"></a>
### 7412-6799b54/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:31322
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 3fb91d7a5717:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... done in 1 minutes and 47 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 22 seconds
(8/14) Running minio-dotnet tests ... done in 36 seconds
(9/14) Running minio-go tests ... done in 48 seconds
(10/14) Running minio-java tests ... FAILED in 21 seconds
{
  "name": "minio-java",
  "function": "listObjects(final String bucketName, final String prefix, final boolean recursive)",
  "args": "prefix :minio, recursive: true",
  "duration": 15735,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code=InternalError, message=We encountered an internal error, please try again., bucketName=minio-java-test-2kp26jt, objectName=minio-java-test-2m01pl1, resource=/minio-java-test-2kp26jt/minio-java-test-2m01pl1, requestId=1591EB7859FD58A8, hostId=08ff086c-62e9-49c3-a1c2-42c76277bbdb)\nrequest={method=PUT, url=http://72.28.97.55:31322/minio-java-test-2kp26jt/minio-java-test-2m01pl1, headers=Content-Type: application/octet-stream\nContent-Encoding: aws-chunked\nx-amz-decoded-content-length: 1\nHost: 72.28.97.55:31322\nUser-Agent: Minio (amd64; amd64) minio-java/dev\nx-amz-content-sha256: STREAMING-AWS4-HMAC-SHA256-PAYLOAD\nx-amz-date: 20190403T090408Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20190403/us-east-1/s3/aws4_request, SignedHeaders=content-encoding;host;x-amz-content-sha256;x-amz-date;x-amz-decoded-content-length, Signature=*REDACTED*\n}\nresponse={code=500, headers=Accept-Ranges: bytes\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: Minio/DEVELOPMENT.2019-04-03T08-46-38Z\nVary: Origin\nX-Amz-Request-Id: 1591EB7859FD58A8\nX-Minio-Deployment-Id: 08ff086c-62e9-49c3-a1c2-42c76277bbdb\nX-Xss-Protection: 1; mode=block\nDate: Wed, 03 Apr 2019 09:04:08 GMT\nTransfer-Encoding: chunked\n}\n >>> [io.minio.MinioClient.executeReq(MinioClient.java:1190), io.minio.MinioClient.execute(MinioClient.java:1056), io.minio.MinioClient.executePut(MinioClient.java:1391), io.minio.MinioClient.executePut(MinioClient.java:1413), io.minio.MinioClient.putObject(MinioClient.java:3986), io.minio.MinioClient.putObject(MinioClient.java:4029), io.minio.MinioClient.putObject(MinioClient.java:3570), FunctionalTest.listObject_test5(FunctionalTest.java:1462), FunctionalTest.runTests(FunctionalTest.java:2985), FunctionalTest.main(FunctionalTest.java:3121)]"
}

Executed 9 out of 14 tests successfully.
```
<a name="4a03f820-0de2-4e1c-b4d8-a3a78001956d"></a>
### 7412-6799b54/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:31323
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e14e94b863ab:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-5d9wyljx9lwplp",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 238,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.59:31323/aws-sdk-go-test-5d9wyljx9lwplp: dial tcp 72.28.97.59:31323: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Can you check why CI failed ?

**Review by poornas — APPROVED** [review]
LGTM

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#32147550-f8f9-4f7d-9262-13e9b9927c9a) |
| mint-fs.sh | :x: [more...](#7d3f16c9-9a4a-4aa1-a898-afec69c4ef2f) |
<a name="7d3f16c9-9a4a-4aa1-a898-afec69c4ef2f"></a>
### 7412-b7e1f5e/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.60:31953
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d222bbee8411:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-oxvib3i3jp0e69",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 214,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.60:31953/aws-sdk-go-test-oxvib3i3jp0e69: dial tcp 72.28.97.60:31953: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="32147550-f8f9-4f7d-9262-13e9b9927c9a"></a>
### 7412-b7e1f5e/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:32177
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d80237274fd0:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-t3wehwifls2f0w",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 261,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.55:32177/aws-sdk-go-test-t3wehwifls2f0w: dial tcp 72.28.97.55:32177: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
ping @poornas

**Pull request merged**